src/org/opensolaris/opengrok/history/Repository.java

```diff
@@ -364,6 +364,8 @@ final void createCache(HistoryCache cache, String sinceRevision)
      */
     abstract String determineBranch() throws IOException;
 
+    abstract String determineCurrentVersion() throws IOException;
```

Comment:
Good point... I followed the determineBranch function (without thinking about it) but you are absolutely right...